library/js/oeUI/insurance/EditPolicyScreenController.js

```diff
@@ -0,0 +1,910 @@
+        this.setupInitialInsurance();
+        this.render();
+    }
+
+    #setupWindowEvents() {
+        let focusInHandler = (evt) => {
+            if (evt.target.nodeName == 'INPUT' || evt.target.nodeName == 'SELECT') {
+                this.__currentFocusedInputName = evt.target.name;
+            }
+        };
+        window.addEventListener('focusin', focusInHandler);
+        this.#boundSetupEvents.push({node: window, event: 'focusin', callback: focusInHandler});
+        // by doing things this way we abstract the communication from the patient selection piece to this controller
+        // we also are safer by grabbing the patient uuid and insurance uuid and refetching just to make sure we have
+        // the right data.
+        let messageHandler = (evt) => {
+            if (event.origin !== window.location.origin) {
+                return; // we only receive events from our same domain
+            }
+            if (evt.data && evt.data.hasOwnProperty('action')) {
+                if (evt.data.action == 'insurance-patient-browser-selected') {
+                    let patientUuid = evt.data.patientUuid || null;
+                    let insuranceUuid = evt.data.insuranceUuid || null;
+                    if (!patientUuid || !insuranceUuid) {
+                        alert(window.top.xl("No patient was selected to copy values from."));
+                        return;
+                    }
+                    // this.__openedSubWindows.forEach(w => w.close()); // close all of the sub windows, not sure why the window.close doesn't work from inside the sub window but we'll do it this way.
+                    let patientInsuranceObject = this.__insurancePolicyService.getPatientInsuranceData(patientUuid, insuranceUuid)
+                        .then(result => {
+                            // copy that data into the currently selected insurance
+                            // make sure we don't overwrite the data that is specific to this insurance
+                            // TODO: perhaps make a populateProtectedFields method that will skip over protected data fields
+                            result.id = this.selectedInsurance.id;
+                            result.uuid = this.selectedInsurance.uuid;
+                            result.type = this.selectedInsurance.type;
+                            result.pid = this.selectedInsurance.pid;
+                            result.puuid = this.selectedInsurance.puuid;
+                            this.selectedInsurance.populate(result);
+                            this.#updateSelectedInsuranceObject(this.selectedInsurance);
+                            this.render();
+                        })
+                        .catch(error => {
+                            console.error(error);
+                            alert(window.top.xl("An error occurred while loading the insurance policy.  Please try again or contact your system administrator."));
```

Comment:
only use single spacing in translated constants

Reply:
Fixed.